apps/web/src/services/analytics/user-attributes.ts

```diff
@@ -0,0 +1,173 @@
+ * @param addedSafes All added Safes across chains
+ * @param currentChainName Current chain name as fallback
+ * @returns Array of network names
+ */
+const getUserNetworks = (addedSafes: Record<string, any>, currentChainName: string): string[] => {
+  try {
+    if (!addedSafes || typeof addedSafes !== 'object') return [currentChainName.toLowerCase()]
+
+    const chainIds = Object.keys(addedSafes)
+    if (chainIds.length === 0) return [currentChainName.toLowerCase()]
+
+    // For now, return the current chain name as the primary network
+    // TODO: Implement proper chain name mapping when needed
+    return [currentChainName.toLowerCase()]
+  } catch (error) {
+    console.warn('Error getting user networks:', error)
+    return [currentChainName.toLowerCase()]
+  }
+}
+
+/**
+ * Gets nested Safe IDs from the Safe configuration
+ * @returns Array of nested Safe addresses
+ */
+const getNestedSafeIds = (): string[] => {
+  // Check if any owners are actually Safe contracts
+  // In a real implementation, we'd need to check if an owner address is a Safe contract
+  // For now, we'll return an empty array as this requires additional API calls
+  // TODO: Implement nested Safe detection
+  return []
+}
+
+/**
+ * Custom hook to compute Safe user attributes for MixPanel tracking
+ * @param safeInfo Current Safe information
+ * @param walletAddress Connected wallet address
+ * @returns SafeUserAttributes object
+ */
+export const useSafeUserAttributes = (
+  safeInfo: ExtendedSafeInfo | null,
+  walletAddress: string | undefined,
+): SafeUserAttributes | null => {
+  const currentChain = useCurrentChain()
+  const addedSafes = useAppSelector(selectAllAddedSafes)
+  const txHistory = useAppSelector(selectTxHistory)
+
+  return useMemo(() => {
+    try {
+      if (!safeInfo || !walletAddress || !currentChain) return null
+
+      const safeAddress = safeInfo.address?.value
+      if (!safeAddress) return null
+
+      const networks = getUserNetworks(addedSafes, currentChain.chainName)
+      const lastTxTimestamp = getLastTxTimestamp(txHistory?.data)
+      const nestedSafeIds = getNestedSafeIds()
+
+      // Parse creation date with fallback
+      let createdAt: Date
+      try {
+        if (safeInfo.implementation?.value) {
+          // Try to parse as unix timestamp (seconds)
+          const timestamp = parseInt(safeInfo.implementation.value, 10)
+          if (!isNaN(timestamp) && timestamp > 0) {
+            createdAt = fromUnixTime(timestamp)
+          } else {
+            createdAt = new Date()
+          }
+        } else {
+          createdAt = new Date()
+        }
+      } catch (error) {
+        createdAt = new Date()
+      }
+
+      const userAttributes: SafeUserAttributes = {
+        safe_id: safeAddress,
+        created_at: createdAt,
+        safe_version: safeInfo.version || '1.0.0',
+        num_signers: safeInfo.owners?.length || 0,
+        threshold: safeInfo.threshold || 1,
```

Comment:
Further up we return early if `safeInfo` doesn't exist. Do we still need these fallbacks?